### ISSUE TITLE
Adversarial neural network MCMC

### ISSUE THREAD

**Issue by ben18785** [pints-team/pints#1040]
[These](https://arxiv.org/pdf/1706.07561.pdf) train adversarial neural networks iteratively to sample from the posterior.

More widely, I think it could be good to have a PINTS branch/offshoot that depends on Tensorflow so we can include these types of samplers.

**Comment by MichaelClerx** [comment]
That's a cool idea! Might belong in the big compare-everything-with-everything project more than PINTS itself? Or do you think we need an intermediary project?

```
PINTS (clean)
   |
performance-testing
```

or 

```
PINTS (clean)
   |
PINTS w Tensorflow
   |
performance-testing
```

?

**Comment by ben18785** [comment]
Yeah, not sure. Think we'd probably want a repo for PINTS w Tensorflow
since there are increasingly more samplers that fit this category (and
something I'd like to experiment with myself!).
